### PR TITLE
Remove Tools::addonsRequest('module') calls

### LINKED ISSUE #24750: Remove Tools::addonsRequest('module') calls
#### Is your feature request related to a problem?

Remove all calls related to `Tools::addonsRequest('module')` 

#### Describe the solution you'd like

Remove all calls, and if needed backports changes into ps_mbo module

### PR DESCRIPTION
**Pull request by Progi1984**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Remove Tools::addonsRequest('module') calls
| Type?             | refacto
| Category?         | CO
| BC breaks?        | yes
| Deprecations?     | no
| Fixed ticket?     | Fixes #24750
| How to test?      | 

**BC Breaks:**
* Removed method `addonsRequest` in class `Tools`
* Removed method `getModuleAttributesById` in class `PrestaShop\PrestaShop\Adapter\Module\AdminModuleDataProvider`
* Removed method `getModuleById` in class `PrestaShop\PrestaShop\Core\Addon\Module\ModuleRepository`
* Removed method `getModuleCartAction` in class `PrestaShopBundle\Controller\Admin\Improve\ModuleController`
* Removed method `getModule` in class `PrestaShopBundle\Service\DataProvider\Marketplace\ApiClient`
* Removed route `admin_module_cart`
* Removed template `src/PrestaShopBundle/Resources/views/Admin/Module/Includes/modal_read_more.html.twig`

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/27111)

### CODE REVIEW

**Comment by Progi1984** [comment]
@atomiix It's rebased.

File: classes/Tools.php [hunk 1]
```diff
@@ -3918,57 +3918,6 @@ public static function cleanNonUnicodeSupport($pattern)
         return $characterCleaner->cleanNonUnicodeSupport($pattern);
     }
 
-    protected static $is_addons_up = true;
-
-    public static function addonsRequest($request, $params = [])
```
Comment (atomiix) [inline classes/Tools.php:3923]: I think this method is still used in [`ps_themecusto`](https://github.com/PrestaShop/ps_themecusto/blob/dev/controllers/admin/AdminPsThemeCustoConfiguration.php#L534), maybe a PR on the module `ps_themecusto` should be made first?

Comment (Progi1984) [reply classes/Tools.php:3923]: Thanks. I will do it this afternoon.

Comment (Progi1984) [reply classes/Tools.php:3923]: @atomiix Done in PrestaShop/ps_themecusto#47

**Review by atomiix — DISMISSED** [review]

**Comment by Progi1984** [comment]
@atomiix I have just rebased :)

**Review by PierreRambaud — DISMISSED** [review]

**Review by atomiix — DISMISSED** [review]

**Comment by Progi1984** [comment]
@atomiix & @PierreRambaud Rebased for @boubkerbribri

**Comment by florine2623** [comment]
Hello @Progi1984 ,

I've checked Modules pages in BO ✅ 

Thanks to @boubkerbribri 's tests auto, he found out these warnings :

- Customer Groups > Edit
Screenshot 2022-01-13 at 17 58 53


- Stores
Screenshot 2022-01-13 at 17 58 39

Could you check ?
Thanks!

**Comment by atomiix** [comment]
@florine2623 I don't think those warnings are related to this PR 🤔

**Comment by boubkerbribri** [comment]
> @florine2623 I don't think those warnings are related to this PR thinking

It's weird because we don't have the problem on develop branch.

**Comment by boubkerbribri** [comment]
> Hello @Progi1984 ,
> 
> I've checked Modules pages in BO white_check_mark
> 
> Thanks to @boubkerbribri 's tests auto, he found out these warnings :
> 
>     * Customer Groups > Edit
> 
> 
> Screenshot 2022-01-13 at 17 58 53
> 
>     * Stores
> 
> 
> Screenshot 2022-01-13 at 17 58 39
> 
> Could you check ? Thanks!


We have also the problem on multistore pages
image

**Comment by boubkerbribri** [comment]
I deleted the comments since it's not related (and need further investigation).
And if not, i think that this PR is ready to be merged.

**Review by PierreRambaud — APPROVED** [review]

**Review by atomiix — APPROVED** [review]

**Comment by Progi1984** [comment]
Thanks @boubkerbribri

**Pull request merged**